Add better handling of lines that do not match the expected format

Anaphory opened the pull request:
Throw error messages when lines don't match the expectation.
Do that at the time when the line is parsed, not at the time when we suddenly stumble over some `undefined` value in `data`.

There is scope to improve this error handling, but it already picks up on having the wrong separator (`,` instead of `\t` in current edictor) and on having rogue strings containing line breaks (but no additional separators).

LinguList commented on js/wordlist.js:
```diff
@@ -549,7 +549,7 @@ function createSelectors() {
 
 /* major function for displaying the Wordlist panel of the Edictor */
-function showWLS(start)
+function showWLS(start, separator="\t")
```
wait a second: you cannot handle defaults in this way, can you? This should be:
```js
function showWLS(start, separator) {
  if (typeof separator == 'undefined') {separator = "\t"}
...
}
```

Anaphory replied:
I remember people discussing that kind of defaults recently. I think the conclusion was that all reasonable browsers support it, Internet Explorer does not. Feel free to change it, it looked like it did work on my browsers.

Anaphory replied:
[Also, I had meant to push that commit only *after* you merging the pull request. Sorry.]

LinguList commented:
Okay, I enhanced this, as we don't look for "," instead of "\t" but also for too many '\t" etc. Also note that we can't use the "trim" function, since we need to allow for empty data values as well, and this would merge them, if the last field, e.g., one with a NOTE is empty...

Anaphory commented:
Oh, did I leave a `separator=","` somewhere? I thought I fixed that after my initial tests.

I was thinking of checking that the number of `sep`s is the same as in the first line everywhere, but then I thought you permit empty cells at the end of a line to be absent (which is also why I thought that trimming would be unproblematic), so I dropped that idea.
